pyro/distributions/torch_distribution.py

```diff
@@ -64,6 +64,35 @@ def shape(self, sample_shape=torch.Size()):
         """
         return sample_shape + self.batch_shape + self.event_shape
 
+    def _sample_shape(self, batch_shape):
+        """
+        Returns the sample shape that must be added to the left of a
+        distribution's existing
+        :attr:`~torch.distributions.distribution.Distribution.batch_shape`,
+        so as to change to the new `batch_shape`.
+
+        :param batch_shape: new batch shape
+        :return: sample shape corresponding to the leftmost dims
+        """
+        batch_shape = list(batch_shape)
+        if len(batch_shape) < len(self.batch_shape):
+            raise ValueError("Expected len(batch_shape) >= len(self.batch_shape), "
+                             "actual {} vs {}".format(len(batch_shape), len(self.batch_shape)))
+        # check sizes of existing dims
+        for dim in range(-1, -1 - len(self.batch_shape), -1):
+            if batch_shape[dim] == -1:
+                batch_shape[dim] = self.batch_shape[dim]
+            elif batch_shape[dim] != self.batch_shape[dim]:
+                if self.batch_shape[dim] != 1:
+                    raise ValueError("Cannot broadcast dim {} of size {} to size {}".format(
+                        dim, self.batch_shape[dim], batch_shape[dim]))
+                else:
+                    raise ShapeMismatchError("The rightmost dimensions of the existing batch " +
```

Comment:
Hmm it seems like we're doing a lot of work before raising this `ShapeMismatchError`. I would have guessed that a simple check would have worked, something like
```py
def expand(self, shape):
    shape = torch.Size(shape)
    cut = len(shape) - len(self.batch_shape)
    left, right = shape[:cut], shape[cut:]
    if right == self.batch_shape:
        return self.expand_by(left)
    # ... continue with existing expand() implementation...
```
Can you help me se what subtlety this simple solution misses?

Reply:
This was mostly just migrated as is from the existing `expand` implementation, but I think your simpler solution will work great and we don't need the more involved implementation (now)! Will change.